parsl/executors/high_throughput/process_worker_pool.py

```diff
@@ -590,6 +590,23 @@ def worker(
             my_cores = avail_cores[cores_per_worker * cpu_worker_id:cores_per_worker * (cpu_worker_id + 1)]
         elif cpu_affinity == "alternating":
             my_cores = avail_cores[worker_id::pool_size]
+        elif cpu_affinity[0:4] == "list":
+            thread_ranks = cpu_affinity.split(":")[1:]
+            if len(thread_ranks) != pool_size:
+                raise ValueError("Affinity list {} has wrong number of thread ranks".format(cpu_affinity))
```

Comment:
This value error (and I guess the other value errors in the affinity code around here) doesn't result in a very good user failure: the worker dies and the manager automatically restarts it (because of worker watchdogs, #2974), forever.

For a workflow user that looks like a hang.

This isn't something that is introduced by this PR, because there were already value errors in this code block that give the same behaviour.

This is basically issue #2072 I think, but I'm crossreferencing it here because I just encountered it, and I think it's more likely to happen with users giving much more complicated inputs here.

Reply:
Wow, that's a gnarly bug. Maybe out-of-scope for this PR?

Reply:
yes probably out of scope for this PR - but users will encounter it more so you should be aware of it more than usual as you encourage users to use this feature.